opentelemetry-etw-metrics/src/exporter/mod.rs

```diff
@@ -69,30 +71,34 @@ fn emit_export_metric_service_request(
 }
 
 impl PushMetricExporter for MetricsExporter {
-    async fn export(&self, metrics: &mut ResourceMetrics) -> OTelSdkResult {
+    async fn export(&self, metrics: &ResourceMetrics) -> OTelSdkResult {
         let schema_url: String = metrics
-            .resource
+            .resource()
             .schema_url()
             .map(Into::into)
             .unwrap_or_default();
 
         // TODO: Reuse this vec across exports by storing it in `MetricsExporter`
         let mut encoding_buffer = Vec::new();
 
-        for scope_metric in &metrics.scope_metrics {
-            for metric in &scope_metric.metrics {
-                let proto_data: Option<TonicMetricData> = metric.data.as_any().try_into().ok();
+        for scope_metric in metrics.scope_metrics() {
+            for metric in scope_metric.metrics() {
+                let proto_data: TonicMetricData = match metric.data() {
+                    AggregatedMetrics::F64(data) => data.into(),
+                    AggregatedMetrics::I64(data) => data.into(),
+                    AggregatedMetrics::U64(data) => data.into(),
+                };
 
                 // This ExportMetricsServiceRequest is created for each metric and will hold a single data point.
                 let mut export_metrics_service_request = ExportMetricsServiceRequest {
                     resource_metrics: vec![TonicResourceMetrics {
-                        resource: Some((&metrics.resource).into()),
+                        resource: Some((metrics.resource()).into()),
```

Comment:
I think we can create the Resource part once per export, instead of once per Metric, as it'll be same for entire batch for metric. (Ideally this can be even re-used across exports, as Resource is fixed!). It may be easier to do this when consolidating user-events and etw encoding into a common helper.